presto-main/src/main/java/io/prestosql/metadata/MetadataUtil.java

```diff
@@ -134,21 +142,28 @@ public static CatalogSchemaName createCatalogSchemaName(Session session, Node no
         return new CatalogSchemaName(catalogName, schemaName);
     }
 
-    public static QualifiedObjectName createQualifiedObjectName(Session session, Node node, QualifiedName name)
+    public static QualifiedObjectName createQualifiedObjectName(Session session, Node node, QualifiedName name, Metadata metadata)
```

Comment:
Instead of passing `Metadata`, let's change it to a `NameCanonicalizer` interface with a single method.